kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/ConflictChecker.java

```diff
@@ -109,6 +114,7 @@ public TransactionRebaseState resolveConflicts(Engine engine) throws ConcurrentW
             handleProtocol(batch.getColumnVector(PROTOCOL_ORDINAL));
             handleMetadata(batch.getColumnVector(METADATA_ORDINAL));
             handleTxn(batch.getColumnVector(TXN_ORDINAL));
+            handleDomainMetadata(batch.getColumnVector(DOMAIN_METADATA_ORDINAL));
```

Comment:
> Also inside the handleDomainMetadata we have addition code that checks there are no duplicates of domain metadata within the batch - which is not correct as the commit file could still contain the duplicates across the batches.

Thanks for catching that! Checking for DM duplicates within each batch is indeed problematic. But I’m considering whether we can remove this duplicate check here entirely. Given that we already validate duplicates on the write path, do you think it’s necessary to also do this check when processing actions from winning commits? I’d love to hear your thoughts on this!

> One suggestion is initialize the Map of the domain metadata before reading each commit and use that in handleDomainMetadata to check for duplicates.

From the code, it appears we’re consuming action batches from all winning commits together. I'm unclear how we would know when a new winning commit is read to initialize a new map of domain metadata for duplicate checks. 

As for domain metadata conflict resolution, I think it doesn't matter whether it can see the entire winning commit file.  Each DM action in the current losing transaction only needs to see a single (or no) corresponding DM action for the same domain in the winning commit.

Reply:
I think we can assume that any successful commit to a Delta table is a valid commit (there are no duplicates). Especially since here, if there are duplicates, it doesn't really affect the outcome of this specific check (looking for conflicts between transactions).

Reply:
So I think just checking for the domain metadata conflict resolution should be sufficient

Reply:
That makes sense to me. So for now I've removed the duplicate check when reading domain metadatas for conflict resolving. This should be correct now.